add new service: paas; PaaS Service: add resource and datasource

PixarV opened the pull request.

Commit:
gha,infra,internal,names: add paas service to names, generate infra

allowed-subcategories.txt changes are exclude and will be added to
another PR.

Commit:
paas: add base types for service definition and schema building

Commit:
paas: implement service datasource

Commit:
paas: add elasticsearch definition and schema

Commit:
paas: add memcached definition and schema

Commit:
paas: add postgresql definition and schema

Commit:
paas: add redis definition and schema

Commit:
paas: implement service resource

Commit:
provider: add paas service resource and datasource

PixarV commented on internal/service/paas/wait.go:
```diff
@@ -0,0 +1,85 @@
+
+func waitServiceCreated(ctx context.Context, conn *paas.PaaS, id string, timeout time.Duration) (*paas.Service, error) {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{ServiceStatusPending, ServiceStatusClaimed, ServiceStatusProvisioning},
```
@astroline, посмотри, пожалуйста, статусную модель создания, обновления и удаления сервиса.

astroline replied:
На текущий момент все ок. В будущем добавятся еще статусы, но их понятное дело будем потом обсуждать

Commit:
paas: add computed parameters: user.id, database.id, update api names

Commit:
paas: add logging and monitoring blocks, modify parameters via api

Additional refactoring:
- inline one-time used methods
- rename getX methods to noun-like
- add condition for ModifyService call: one of the [`backup_setting`,
`<service>.user`, `<service>.database`] needs to be changed
- add default implementations for service parameters schema building and
 convertation methods
- add computed parameter `vpc_id`

Commit:
paas: pgsql: add dimension to fields containing bytes when send to api
